### PR TITLE
added logger

### PR DESCRIPTION
**Pull request by matviichuk92**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -26,6 +26,12 @@
             <artifactId>hibernate-java8</artifactId>
             <version>5.4.5.Final</version>
         </dependency>
+        <dependency>
+            <groupId>log4j</groupId>
+            <artifactId>log4j</artifactId>
+            <version>1.2.17</version>
+        </dependency>
+
```
Comment (AnaLanda) [inline pom.xml]: I think you can remove this empty line)

File: src/main/java/com/cinema/Application.java [hunk 2]
```diff
@@ -66,11 +68,11 @@ public static void main(String[] args) throws AuthenticationException {
...
-        shoppingCartService.addSession(session,romaFromDB);
-        shoppingCartService.addSession(secondSession,romaFromDB);
+        shoppingCartService.addSession(session, romaFromDB);
+        shoppingCartService.addSession(secondSession, romaFromDB);
         orderService.completeOrder(romaFromDB, shoppingCartService
                 .getByUser(romaFromDB).getTickets());
-        System.out.println("Empty cart : " + shoppingCartService.getByUser(romaFromDB));
-        System.out.println(orderService.getOrderHistory(romaFromDB));
+        logger.debug("Empty cart : " + shoppingCartService.getByUser(romaFromDB));
+        logger.debug(orderService.getOrderHistory(romaFromDB));
```
Comment (AnaLanda) [inline src/main/java/com/cinema/Application.java]: I think Ksenia said in the Q&A that we can use logger.info in this project)

File: src/main/java/com/cinema/dao/impl/CinemaHallDaoImpl.java [hunk 3]
```diff
@@ -21,6 +22,7 @@ public CinemaHall add(CinemaHall cinemaHall) {
             transaction = session.beginTransaction();
             session.save(cinemaHall);
             transaction.commit();
+            Application.logger.info(cinemaHall);
```
Comment (AnaLanda) [inline src/main/java/com/cinema/dao/impl/CinemaHallDaoImpl.java]:
```suggestion
            Application.logger.info("Successfully added " + cinemaHall + " to DB");
```
You can also create a `Logger` for each Dao without importing `Application`. If you have a problem running the code then, you can replace the old injector with the new one (see the comments for today's Q&A with Ksenia)

File: src/main/java/com/cinema/dao/impl/MovieDaoImpl.java [hunk 4]
```diff
@@ -21,6 +22,7 @@ public Movie add(Movie movie) {
             transaction = session.beginTransaction();
             session.save(movie);
             transaction.commit();
+            Application.logger.info(movie);
```
Comment (AnaLanda) [inline src/main/java/com/cinema/dao/impl/MovieDaoImpl.java]: Maybe you can make all logger messages more informative)

File: src/main/resources/log4j.properties [hunk 5]
```diff
@@ -0,0 +1,13 @@
+log4j.rootLogger=DEBUG, stdout, file
```
Comment (AnaLanda) [inline src/main/resources/log4j.properties:1]: Maybe you can also add this for Hibernate logging: 

```
log4j.logger.org.hibernate=INFO
log4j.logger.org.hibernate.SQL=DEBUG
```

**Review by AnaLanda — COMMENTED** [review]
Looks good!

File: src/main/resources/log4j.properties [hunk 6]
```diff
@@ -0,0 +1,13 @@
...
+
+log4j.appender.stdout=org.apache.log4j.ConsoleAppender
+log4j.appender.stdout.Target=System.out
+log4j.appender.stdout.layout=org.apache.log4j.PatternLayout
+log4j.appender.stdout.layout.ConversionPattern=%d{yyyy-MM-dd HH:mm:ss} %-5p %c{1}:%L - %m%n
+
+log4j.appender.file=org.apache.log4j.RollingFileAppender
+log4j.appender.file.File=C:\\log4j-application.log
```
Comment (KseniiaMakarova) [inline src/main/resources/log4j.properties]: let's place this file to your `logs` folder

File: src/main/java/com/cinema/Application.java [hunk 7]
```diff
@@ -14,8 +14,10 @@
 import com.cinema.service.ShoppingCartService;
 import com.cinema.service.UserService;
 import java.time.LocalDateTime;
+import org.apache.log4j.Logger;
 
 public class Application {
+    public static final Logger logger = Logger.getLogger(Application.class);
```
Comment (KseniiaMakarova) [inline src/main/java/com/cinema/Application.java]:
```suggestion
    private static final Logger logger = Logger.getLogger(Application.class);
```

File: src/main/java/com/cinema/Application.java (hunk 2 above)
Comment (KseniiaMakarova) [reply src/main/java/com/cinema/Application.java]: Yep, I think it is better to use `info` level here
`debug` is for more fine-grained logging

File: src/main/java/com/cinema/dao/impl/CinemaHallDaoImpl.java (hunk 3 above)
Comment (KseniiaMakarova) [reply src/main/java/com/cinema/dao/impl/CinemaHallDaoImpl.java]: I totally agree
please do not import Logger - create it in each DaoImpl class
use the updated Injector if needed

Comment (KseniiaMakarova) [inline src/main/java/com/cinema/dao/impl/CinemaHallDaoImpl.java]: let's add more informative messages as Anastasiya suggests

File: src/main/java/com/cinema/dao/impl/ShoppingCartImpl.java [hunk 8]
```diff
@@ -25,6 +26,7 @@ public ShoppingCart add(ShoppingCart shoppingCart) {
             transaction = session.beginTransaction();
             session.save(shoppingCart);
             transaction.commit();
+            Application.logger.info(shoppingCart);
```
Comment (KseniiaMakarova) [inline src/main/java/com/cinema/dao/impl/ShoppingCartImpl.java]: let's also log `update` operation

File: src/main/java/com/cinema/security/AuthenticationServiceImpl.java [hunk 9]
```diff
@@ -21,6 +22,7 @@ public User login(String email, String password) throws AuthenticationException
         Optional<User> userFromDB = userService.findByEmail(email);
         if (userFromDB.isPresent() && HashUtil.hashPassword(password, userFromDB.get().getSalt())
                 .equals(userFromDB.get().getPassword())) {
+            Application.logger.info(userFromDB.get());
```
Comment (KseniiaMakarova) [inline src/main/java/com/cinema/security/AuthenticationServiceImpl.java]: add more informative messages everywhere

File: src/main/java/com/cinema/Application.java [hunk 10]
```diff
@@ -66,11 +68,11 @@ public static void main(String[] args) throws AuthenticationException {
 
         authenticationService.register(roma.getEmail(), roma.getPassword());
         User romaFromDB = authenticationService.login(roma.getEmail(), roma.getPassword());
```
Comment (KseniiaMakarova) [inline src/main/java/com/cinema/Application.java]: let's wrap call to `login()` method in try-catch and add log.warn with the informative message in `catch` block

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

**Review by KseniiaMakarova — APPROVED** [review]

**Pull request merged**